code/game/atoms_movable.dm

```diff
@@ -329,4 +329,9 @@
 	. = ..()
 
 /atom/movable/CanPass(atom/movable/mover, turf/target, height=1.5, air_group = 0)
-	. = ..() || (mover && !(mover.movable_flags & MOVABLE_FLAG_NONDENSE_COLLISION) && !mover.density)
+	. = ..()
+	if(. && mover)
```

Comment:
This does actually change the logic: it used to be that if `..()` is true, we return true and continue on; the condition is an alternative success condition. The new logic is that there are additional fail conditions. In particular, in the old logic, all nondense movable movers would succeed unless the  flag is set, and that happens right here (..() will be false if src is dense, for example).

Generally, if CanPass returns false on a movable on a turf, turf/Enter will return false, which will prevent the move and step will fail.